### PR TITLE
Cleanup in Split mask propagation transformation

### PR DESCRIPTION
**Pull request by mateusztabaka**

### CODE REVIEW

**Comment by daniil-lyakhov** [comment]
@mateusztabaka, thanks for the PR! LGTM in general, I have a couple minor comments

File: src/common/offline_transformations/src/pruning/propagate_masks.cpp [hunk 1]
```diff
@@ -1432,6 +1432,42 @@ class ngraph::pass::mask_propagation::Transpose : public MatcherPass {
     }
 };
 
+static ngraph::Mask::Ptr create_split_mask(ngraph::Mask::Ptr input_mask,
+                                           int64_t axis,
+                                           uint64_t split_start,
+                                           uint64_t split_end) {
```
Comment (daniil-lyakhov) [inline src/common/offline_transformations/src/pruning/propagate_masks.cpp:1438]: Please consider adding a comment like here
https://github.com/openvinotoolkit/openvino/blob/4d11c37636eab540f33923ce7f8aa0aee15566e9/src/common/offline_transformations/src/pruning/propagate_masks.cpp#L1091-L1094.

I would recommend to reflect meaning of this function in its name: it isn't just creating split mask but it's adding callbacks to input/output masks besides. I believe that name like `create_connect_split_output_mask` would gain code simplicity. @mateusztabaka, what do you think?

Comment (daniil-lyakhov) [inline src/common/offline_transformations/src/pruning/propagate_masks.cpp:1438]: Just to make sure
```suggestion
                                           const int64_t axis,
                                           const uint64_t split_start,
                                           const uint64_t split_end) {
```

Comment (mateusztabaka) [reply src/common/offline_transformations/src/pruning/propagate_masks.cpp]: done

Comment (mateusztabaka) [reply src/common/offline_transformations/src/pruning/propagate_masks.cpp]: done

**Review by daniil-lyakhov — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

**Pull request merged**